### ISSUE TITLE
Generic physics engine API?

### ISSUE THREAD

**Issue by traverseda** [ValkyrienSkies/Valkyrien-Skies#388]
Valkyrien skies aims to be a very complete mod. It would be nice if instead it was two mods, a generic "phsyics core" style mod and a different mod that adds all the recipes and custom ores and stuff.

The reason for this is that Valkyrien Skies has a very particular "flavour" that doesn't necessarily fit all settings, or all difficulty curves. Server operators might what a version where it's much easier to make airships, or they might want to integrate the airships with some particular modpack. I can see specific mods having specific ways of making airships. Like thaumcraft could have a "thaumic resonator" powered airship or something.

The core physics engine of this is too cool to be limited to just one mod, making it a resource other mods can draw upon could make some awesome stuff. Also exasperating out the concerns like that could save a lot of dev work, because the more normal mod stuff like "what recipes should this use" or "how do we control this" is a lot easier than the actual physics engine stuff.

**Comment by Rubydesic** [comment]
Yes, this is being actively worked on. It's a very tricky problem that requires a lot of code cleanup and refactoring and has been taking us a long time but don't worry, it's coming. We are trying to optimize the mod as best as possible so that it can be utilized by any mod without the need for our control system or even our physics engine. Hang in there!

**Comment by traverseda** [comment]
Glad to hear that's on the radar. Sounds great.

**Comment by DeltaNedas** [comment]
technically the non-engine content is done via "addons", but not as separate mods (yet)
